### ISSUE TITLE
Loosing series merging when performing a rollup

### ISSUE THREAD

**Issue by AxelDelmas** [influxdata/influxdb#5877]
I'm trying to use continuous queries to perform rollups (trying to achieve something similar to #5335 or #5234)

I'm using something like `CREATE CONTINUOUS QUERY mycq ON mydb BEGIN SELECT mean(value) AS mean_value INTO mydb_5m.fortydays.:MEASUREMENT FROM mydb.oneday./mymeasurement/ GROUP BY time(5m), * END` to aggregate my data. 

Then `SELECT mean_value from mymeasurement` (using `mydb_5m`) returns a data point _per serie_ for every 5m period, where I'd want it to return a single point: the mean of the merged series for that 5m period.
The same happens if we filter only by a subset of `mymeasurement`'s tag set: we get as many points per 5m period as the number of combinations of the tags we didn't specify.

This is solved easily when we aggregate using a sum, as we can sum sums, but it doesn't mean anything anymore if we're trying to combine means or percentiles.

In other words, we loose series merging when performing a rollup, and I can't see an easy way to work around this. All I can think of is to add a continuous query per most-used tag combination, but it can be quite cumbersome.

**Comment by jonseymour** [comment]
Perhaps I am misunderstanding something about your requirements, but what is wrong with:

```
CREATE CONTINUOUS QUERY mycq ON mydb BEGIN SELECT mean(value) AS mean_value INTO mydb_5m.fortydays.:MEASUREMENT FROM mydb.oneday./mymeasurement/ GROUP BY time(5m) END
```

e.g. deleting the ', *' from the end of the series.

If this is going to be different to a mean of means, then you probably need two CQ's one to create a mean of each series and then a second one to take the mean of the aggregate of the series created by the first CQ

The fundamental limitation here, I think, is that InfluxQL doesn't have the equivalent of SQL's nested SELECTS so there are some kinds of aggregation that are not easily expressed in a single InfluxQL query.

If your requirements justify it, you might also look at the Kapacitor tool as it does provide additional flexibility for expressing data processing requirements.

**Comment by AxelDelmas** [comment]
That's right, although if I have 5 tags: `tag1, tag2, tag3, tag4, tag5` and I want to make a SELECT with a WHERE clause only on `tag1` and `tag2` then I also have to create a separate continuous query with `group by time(5m), tag1, tag2`, and so on with every tag combination I expect I'll need.

What I'm suggesting is adding tag combination where some tags are left "undefined" when performing a `group by *`. In the example above, a `SELECT ... WHERE tag1 = 'somevalue' and tag2 = 'othervalue'` would return the value of the serie defined by `tag1='somevalue',tag2='othervalue'`, that is where all the series which share these tag values, and with any value on the other tags, were merged during aggregation.

I understand this would require non trivial changes. For example, for now storing the aggregates is quite simple : one point per "group by period" per series on the same field. We can't expect that to work with the change I'm suggesting.
However I think that change along those lines is necessary to make rollups more usable with InfluxDB, and that's really necessary for application with high point density.
